Extensions/TextObject/TextObject.cpp

```diff
@@ -27,52 +25,116 @@ TextObject::TextObject()
       bold(false),
       italic(false),
       underlined(false),
-      colorR(0),
-      colorG(0),
-      colorB(0),
-      textAlignment("left")
+      color("0;0;0"),
+      textAlignment("left"),
+      isOutlineEnabled(false),
+      outlineThickness(2),
+      outlineColor("255;255;255"),
+      isShadowEnabled(false),
+      shadowColor("0;0;0"),
+      shadowOpacity(127),
+      shadowAngle(90),
+      shadowDistance(4),
+      shadowBlurRadius(2)
 {
 }
 
 TextObject::~TextObject(){};
 
 void TextObject::DoUnserializeFrom(gd::Project& project,
                                    const gd::SerializerElement& element) {
-  SetString(element.GetChild("string", 0, "String").GetValue().GetString());
-  SetFontName(element.GetChild("font", 0, "Font").GetValue().GetString());
-  SetTextAlignment(element.GetChild("textAlignment").GetValue().GetString());
-  SetCharacterSize(element.GetChild("characterSize", 0, "CharacterSize")
+  // Compatibility with GD <= 5.3.188
+  // end of compatibility code
+  bool isLegacy = !element.HasChild("content");
+  auto &content = isLegacy ? element : element.GetChild("content");
```

Comment:
Here, I couldn't find a way to frame the code so I didn't put anything inside it.